src/test/ui/rfc-2011-nicer-assert-messages/feature-gate-generic_assert.rs

```diff
@@ -0,0 +1,184 @@
+// compile-flags: --test
+// edition:2021
+// ignore-tidy-linelength
+// run-pass
```

Comment:
Please also add a test showing various failure cases when the macro is used in invalid ways (either due to types or just wrong syntax).

Reply:
`assert!` parsing already has tests but I added a few more "just in case". Perhaps these newly added syntax tests can be removed in face of a potential stabilization

Reply:
Ah, we could reuse those tests by making them revision tests. One revision without your feature gate and one with. This way we'll see if any differences pop up

Reply:
Done

If you have time, you would mind doing a deeper review? If not, then don't worry.